src/main/java/seedu/address/logic/commands/ArchiveCommand.java

```diff
@@ -0,0 +1,60 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Archives a person identified using it's displayed index from the address book.
+ */
+public class ArchiveCommand extends Command {
+
+    public static final String COMMAND_WORD = "archive";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Archives the person identified by the index number used in the displayed person list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_ARCHIVE_PERSON_SUCCESS = "Archived Person: %1$s";
+
+    private final Index targetIndex;
+
+    /**
+     * Constructs an ArchiveCommand.
+     *
+     * @param targetIndex the index number shown in the displayed person list.
+     */
+    public ArchiveCommand(Index targetIndex) {
+        requireAllNonNull(targetIndex);
```

Comment:
Since it's just one argument, `requireNonNull(targetIndex)` will do.

Reply:
Thanks for pointing out! Will edit it and change it in the next iteration.